src/attr.rs

```diff
@@ -0,0 +1,64 @@
+use std::fmt;
+use std::borrow::{Borrow, Cow};
+
+use escape::Escaped;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// All underscores in the attribute name will be `Display`-ed as hyphens
+pub struct Attribute<'a> {
+    pub name: Cow<'a, str>,
+    pub value: Cow<'a, str>,
+}
+
+impl<'a> Attribute<'a> {
+    pub fn new<N, V>(name: N, value: V) -> Attribute<'a>
+        where N: Into<Cow<'a, str>>,
+              V: Into<Cow<'a, str>>
+    {
+        Attribute {
+            name: name.into(),
+            value: value.into(),
+        }
+    }
+}
+
+impl<'a> fmt::Display for Attribute<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let name: &str = self.name.borrow();
+        write!(f, "{}=\"{}\"", hyphenate(name.into()), Escaped(&self.value))
```

Comment:
I'd prefer to explicitly hyphenate on input, maybe a separate constructor like `new_with_hyphenated_id`, or move it to `mod util` or something and have the macro call it (unfortunately it has to be public for the macro, otherwise I'd suggest making it a private method).